tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -23,22 +23,22 @@ using namespace shogun;
 #include <shogun/mathematics/eigen3.h>
 using namespace Eigen;
 
-//function to create sparse matrix
-template<typename T>
-SGSparseMatrix<T> GenerateSparseMatrix(const int32_t numberOfFeatures,
-                                       const int32_t numberOfVectors,
-                                       const float64_t sparseLevel)
+//function can generate both sparse and dense matrix accroding to sparse convention
+template<class MatrixType>
+void GenerateMatrixAccrodingToSparseConvention(const float64_t &sparseLevel,
+                                               const int32_t   &numberOfFeatures,
+                                               const int32_t   &numberOfVectors,
+                                               const int32_t   &randSeed,
+                                               MatrixType* matrix)
```

Comment:
I don't think it is possible to create a sparse and dense matrix and fill in their value inside the function. since you need to use SparseMatrix(numberOfFeats, numberOfVects) but DenseMatrix(numberOfVecs, numberOfFeats) to initialize. Thus it is only possible to initialize the matrix outside the function and fill in their value inside the function.